Sources/BBCodeParser.php

```diff
@@ -3619,7 +3619,10 @@ protected function closeTags(): void
 			return;
 		}
 
-		$look_for = strtolower(substr($this->message, $this->pos + 2, $pos2 - $this->pos - 2));
+		// @todo: $pos2 could be false?
+		$pos2 = (int) $pos2;
+
+		$look_for = strtolower(substr($this->message, $this->pos + 2, (int) ($pos2 - $this->pos - 2)));
```

Comment:
```suggestion
		$look_for = strtolower(substr($this->message, $this->pos + 2, (int) $pos2 - $this->pos - 2));
```

But moreover, this probably belongs in a separate PR.